launcher/FileSystem.h

```diff
@@ -186,4 +340,231 @@ bool overrideFolder(QString overwritten_path, QString override_path);
+    FAT,
+    NTFS,
+    REFS,
+    EXT,
+    EXT_2_OLD,
+    EXT_2_3_4,
+    XFS,
+    BTRFS,
+    NFS,
+    ZFS,
+    APFS,
+    HFS,
+    HFSPLUS,
+    HFSX,
+    FUSEBLK,
+    F2FS,
+    UNKNOWN
+};
+
+/**
+ * @brief Ordered Mapping of enum types to reported filesystem names
+ * this maping is non exsaustive, it just attempts to capture the filesystems which could be reasonalbly be in use .
+ * all string values are in uppercase, use `QString.toUpper()` or equivalent during lookup.
+ * 
+ * QMap is ordered
+ * 
+ */
+static const QMap<FilesystemType, QString> s_filesystem_type_names = {
+    {FilesystemType::FAT,    QString("FAT")},
+    {FilesystemType::NTFS,   QString("NTFS")},
+    {FilesystemType::REFS,   QString("REFS")},
+    {FilesystemType::EXT,    QString("EXT")},
+    {FilesystemType::EXT_2_OLD,    QString("EXT_2_OLD")},
+    {FilesystemType::EXT_2_3_4,    QString("EXT2/3/4")},
+    {FilesystemType::XFS,    QString("XFS")},
+    {FilesystemType::BTRFS,  QString("BTRFS")},
+    {FilesystemType::NFS,    QString("NFS")},
+    {FilesystemType::ZFS,    QString("ZFS")},
+    {FilesystemType::APFS,   QString("APFS")},
+    {FilesystemType::HFS,    QString("HFS")},
+    {FilesystemType::HFSPLUS, QString("HFSPLUS")},
+    {FilesystemType::HFSX,    QString("HFSX")},
+    {FilesystemType::FUSEBLK, QString("FUSEBLK")},
+    {FilesystemType::F2FS, QString("F2FS")},
+    {FilesystemType::UNKNOWN, QString("UNKNOWN")}
+};
+
+
+/**
+ * @brief Ordered Mapping of reported filesystem names to enum types
+ * this maping is non exsaustive, it just attempts to capture the many way these filesystems could be reported.
+ * all keys are in uppercase, use `QString.toUpper()` or equivalent during lookup.
+ * 
+ * QMap is ordered
+ * 
+ */
+static const QMap<QString, FilesystemType> s_filesystem_type_names_inverse = {
+    {QString("FAT"), FilesystemType::FAT},
+    {QString("NTFS"), FilesystemType::NTFS},
+    {QString("REFS"), FilesystemType::REFS},
+    {QString("EXT2_OLD"), FilesystemType::EXT_2_OLD},
+    {QString("EXT_2_OLD"), FilesystemType::EXT_2_OLD},
+    {QString("EXT2"), FilesystemType::EXT_2_3_4},
+    {QString("EXT3"), FilesystemType::EXT_2_3_4},
+    {QString("EXT4"), FilesystemType::EXT_2_3_4},
+    {QString("EXT2/3/4"), FilesystemType::EXT_2_3_4},
+    {QString("EXT_2_3_4"), FilesystemType::EXT_2_3_4},
+    {QString("EXT"), FilesystemType::EXT}, // must come after all other EXT variants to prevent greedy detection
+    {QString("XFS"), FilesystemType::XFS},
+    {QString("BTRFS"), FilesystemType::BTRFS},
+    {QString("NFS"), FilesystemType::NFS},
+    {QString("ZFS"), FilesystemType::ZFS},
+    {QString("APFS"), FilesystemType::APFS},
+    {QString("HFSPLUS"), FilesystemType::HFSPLUS},
+    {QString("HFSX"), FilesystemType::HFSX},
+    {QString("HFS"), FilesystemType::HFS},
+    {QString("FUSEBLK"), FilesystemType::FUSEBLK},
+    {QString("F2FS"), FilesystemType::F2FS},
+    {QString("UNKNOWN"), FilesystemType::UNKNOWN}
+};
```

Comment:
mostly because I was too lazy to write a lookup loop :P.
I'll change it though.

Reply:
Actually... I'm not sure I *should* go with a list of pairs.

this isn't a one to one mapping.

the OS can report many different strings for some of the filesystems (EXT2, EXT3, EXT4 all share the same magic for example)

the Type to string map is one to one but the string to type map is many to one.

and honestly, the isn't used in a function called in a loop. a single hash look up is kinda of fine?

I'll still do the QStringLiteral change.

Reply:
> this isn't a one to one mapping.

technically, the pair list approach could be a many-to-one mapping if you do a ordered linear search: many filesystem type strings can map to the same type, and a type can map to the first string mapping to it. It could also be achieved with a single map, since it maintains order, but i think a list better exposes the intention of mapping both string -> type and type -> string.

I don't really mind the performance of either approach, i just dislike having two different data structures that need to be managed and maintained separately even though they are intimately related.

Reply:
Getting the many to one mapping from a list isn't the issue really, it's the inversion of the mapping needing a different search order for lookup that gets me.
Id love to deduplicate the structure frankly.

Reply:
ok I think I've found a method I'm satisifed with. `QMap<Enum, QStringList>` where the first elm of the list os the prefered name.

no more duplication.